tools/CMakeLists.txt

```diff
@@ -2,11 +2,14 @@ include(CheckIncludeFiles)
 check_include_files("xpc/xpc.h" HAVE_XPC_H)
 
 if(HAVE_XPC_H AND SWIFT_BUILD_SOURCEKIT)
-  set(BUILD_SOURCEKIT_XPC_SERVICE TRUE)
+  set(BUILD_SOURCEKIT_XPC_SERVICE_DEFAULT TRUE)
 else()
-  set(BUILD_SOURCEKIT_XPC_SERVICE FALSE)
+  set(BUILD_SOURCEKIT_XPC_SERVICE_DEFAULT FALSE)
 endif()
 
+set(BUILD_SOURCEKIT_XPC_SERVICE BUILD_SOURCEKIT_XPC_SERVICE_DEFAULT CACHE BOOL
```

Comment:
This is slightly difficult to read.  I think that introducing some noise here would be preferable.

```suggestion
set(BUILD_SOURCEKIT_XPC_SERVICE ${BUILD_SOURCEKIT_XPC_SERVICE_DEFAULT} CACHE BOOL
```

Reply:
@cwakamosince this is a bool, I think this should be set using option(, not set(, no?

Reply:
Just pushed an updated commit which resolves these two issues.